server/src/com/cloud/servlet/ConsoleProxyServlet.java

```diff
@@ -461,7 +463,12 @@ private String composeConsoleAccessUrl(String rootUrl, VirtualMachine vm, HostVO
             param.setClientTunnelSession(parsedHostInfo.third());
         }
 
-        sb.append("/ajax?token=" + encryptor.encryptObject(ConsoleProxyClientParam.class, param));
+
+        if(webSocketRequest){
```

Comment:
NPE can be checked here. However we should not handle /ajax is we're completely moving to novnc.